### PR TITLE
cmake: add so version to libhypre

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,7 @@ endif ()
 
 # The version number.
 set(HYPRE_VERSION 2.31.0)
+set(HYPRE_SOVERSION 2)
```
Comment (junghans) [inline src/CMakeLists.txt]: On a second thought the `soversion` should maybe be `31` instead.

**Comment by junghans** [comment]
ping @waynemitchell

**Comment by victorapm** [comment]
Hi @junghans, thank you for your contribution! I talked to @streeve last week and pointed out this development branch: https://github.com/hypre-space/hypre/tree/cmake-hip

This branch will include the soversion change that your proposed here. I appreciate if you can test the branch and provide feedback. It's almost complete, except for documentation changes. Thank you!

**Comment by junghans** [comment]
Perfect, I will look!

**Comment by junghans** [comment]
@streeve @victorapm The `cmake-hip` branch would work for me, but given https://github.com/hypre-space/hypre/blob/cmake-hip/INSTALL.md#abi-compatibility, we might not want to use `2` as the so version.

**Comment by victorapm** [comment]
@junghans Thanks for bringing this up. I updated the branch to use `major.minor` as the SOVERSION

**Comment by junghans** [comment]
I have to check but I believe the so version needs to be an integer.

**Comment by junghans** [comment]
Ok, I think CMake actually allows it, but i generated a non-standard symlink hierarchy: 
```
libhypre.so -> libhypre.so.2.32 -> libhypre.so.2.32.0
```

As a normal user I would more expect something like:
```
libhypre.so -> libhypre.so.2 -> libhypre.so.2.32
```
(see https://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html).

But of course as there is no [abi-compatibility](https://github.com/hypre-space/hypre/blob/cmake-hip/INSTALL.md#abi-compatibility), maybe:
```
libhypre.so -> libhypre.so.32
```
or
```
libhypre.so -> libhypre.so.232
```

would be better.

**Comment by victorapm** [comment]
@junghans sounds great. Thanks for digging into this!

**Pull request closed without merging**